src/Hubee.Validation.Sdk/Core/Models/ValidatableSchema.cs

```diff
@@ -0,0 +1,11 @@
+﻿using Hubee.Validation.Sdk.Core.Interfaces;
+
+namespace Hubee.Validation.Sdk.Core.Models
+{
+    public abstract class ValidatableSchema: IValidatableSchema
+    {
+        public ValidationResult ValidationResult { get; set; }
```

Comment:
Tô pensando aqui, da forma que temos (agora e antes), caso façamos isso aqui:

```c#
var cliente = new Cliente(x, y, z);
cliente.ValidationResult.IsValid();
```
Vamos ter null pointer né?

Pensei nessas abordagens:
- No getter do ValidationResult fazer uma verificação se ele é null e subir uma exceção caso for;
- No getter do ValidationResult se ele for null, executar a validação e depois retornar o resultado (se possível).

O que acha? Podemos testar se rola juntos :)

Reply:
Faz todo sentido man, boa observação.

Penso que o uso do ValidadeSchema(); fica mas declarativo que as  validações foram realizadas. Então por esse motivo penso que a primeira abordagem faz mas sentido.

O que você acha?

Reply:
Acho que qualquer uma das duas ajudaria viu... 

Pensando na segunda, seria uma exception a menos pra subir, mas teríamos que executar a validação no getter e não sei se isso rola.

Vou testar aqui.

Reply:
@luizmirandati rolou a auto-validação. Subi um [PR ](https://github.com/HubeePlatform/hubee-validation-sdk/pull/6) que acha?

Reply:
Top demais man só vamo!